src/vs/workbench/services/runtimeStartup/common/runtimeStartup.ts

```diff
@@ -1376,20 +1377,23 @@ export class RuntimeStartupService extends Disposable implements IRuntimeStartup
 			.map(session => {
 				const activeSession =
 					this._runtimeSessionService.getActiveSession(session.metadata.sessionId);
+
 				const metadata: SerializedSessionMetadata = {
+					sessionName: session.dynState.sessionName,
```

Comment:
Since this changes the serialization format, would also recommend changing the name of the storage key so that previously serialized sessions (which are missing this new field) don't cause errors